### ISSUE TITLE
Skill-Time Profile Export - Component 1: Skill-Time Conversion Microservice

### ISSUE THREAD

**Issue by TheMorpheus407** [Bootstrap-Academy/Bootstrap-Academy#23]
This service is responsible for converting the skill level of the user into an estimated time investment for each skill.

#### Functional Requirements:
- [ ] **Data Retrieval:** The service must interface with the User Profile Microservice to fetch the user's current skill levels from the database.  
- [ ] **Conversion Logic:** Implement a mechanism that translates a user's skill level into an estimated time investment. The formula or conversion factor should be adjustable to refine based on accuracy feedback over time.  
- [ ] **Data Updating:** Post conversion, this estimated time data should be stored back in the user's profile within the database. This necessitates interfacing with the User Profile Microservice.

#### Non-Functional Requirements:
- [ ] **Performance:** Ensure the service is optimized for rapid response times to maintain a seamless user experience.  
- [ ] **Reliability:** The service must consistently yield accurate and reliable results.  
- [ ] **Scalability:** Design the service with scalability in mind, allowing it to handle increased loads as the user base expands.

## Acceptance Criteria:
- [ ] Accurately convert user skill levels to a representative measure of time spent learning or practicing that skill.  
- [ ] Maintain consistent conversion across all skill types.  
- [ ] Dynamically update the estimated time as users' skill levels change.  
- [ ] Successfully integrate with user profiles to fetch skill level data in real-time.

#### Dependencies:
- None for this component. However, subsequent components (like the PDF Export Microservice) will depend on the successful completion of this component.

#### Notes:
- As this is the initial phase, there are no UI/UX mockups. After discussions and approvals on this issue, designers will create relevant mockups.  
- The database schemas are pre-established. Any interactions or modifications must align with the current setup.

**Comment by Lauritz-Tieste** [comment]
Did I understand correctly that this PR requires 

1. calculate the estimated time on a SubSkill
Example formula: `AsCompletedMarkedVideoTimes + NumberCompletedQuizzesAndMatchings * NumberAnswersInQuizAndMatching / 4 + NumberCompletedChallenges * 30 minutes `
2. save these values (as total time and time per SubSkill) in the user profile
3. retrieve the data from the user profile (`GET /auth/users/{user_id}`)

It is not required here: UI Implementation

**Comment by TheMorpheus407** [comment]
Indeed, yes. Here's just the Microservice as ticket

**Comment by Lauritz-Tieste** [comment]
And should this be built into `GET /auth/users/{user_id}`?

**Comment by TheMorpheus407** [comment]
No, it's not about authentication, rather it's about their profile

**Comment by Lauritz-Tieste** [comment]
Please assign me

**Comment by Mert-55** [comment]
Hi do I understand this correctly
can the public profile be queried for example via the following route GET `auth/users/profiles/{pub_profile_id}` or am I wrong and need a separate ms for the profiles? Many thanks in advance :)!
